File: src/jp/lionas/alexa/sample/java/CancelandStopIntentHandler.java
package jp.lionas.alexa.sample.java;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;

import java.util.Optional;

import static com.amazon.ask.request.Predicates.intentName;

public class CancelandStopIntentHandler implements RequestHandler {

    @Override
    public boolean canHandle(HandlerInput handlerInput) {
        return handlerInput.matches(intentName("AMAZON.StopIntent").or(intentName("AMAZON.CancelIntent")));
    }

    @Override
    public Optional<Response> handle(HandlerInput handlerInput) {
        return handlerInput.getResponseBuilder()
                .withSpeech("Goodbye")
                .withSimpleCard("HelloWorld", "Goodbye")
                .build();
    }

}

File: src/jp/lionas/alexa/fundra/handler/CancelAndStopIntentHandler.kt
package jp.lionas.alexa.fundra.handler

import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.dispatcher.request.handler.RequestHandler
import com.amazon.ask.model.Response
import com.amazon.ask.request.Predicates.intentName
import jp.lionas.alexa.fundra.def.Message
import java.util.*

class CancelAndStopIntentHandler : RequestHandler {

    override fun canHandle(handlerInput: HandlerInput): Boolean {
        return handlerInput.matches(intentName("AMAZON.StopIntent")
                .or(intentName("AMAZON.CancelIntent")))
    }

    override fun handle(handlerInput: HandlerInput): Optional<Response> {
        return handlerInput.responseBuilder
                .withSpeech(Message.CANCEL_SPEECH_TEXT)
                .withSimpleCard(Message.SKILL_NAME, Message.CANCEL_SPEECH_TEXT)
                .build()
    }

}
